src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -181,42 +181,19 @@ private bool GetCrossgen2TargetOS(out string targetOS)
 
             // Determine targetOS based on target rid.
             // Use the runtime graph to support non-portable target rids.
+            // Use the full target rid instead of just the target OS as the runtime graph
+            // may only have the full target rid and not an OS-only rid for non-portable target rids
+            // added by our source-build partners.
             var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
             string portablePlatform = NuGetUtils.GetBestMatchingRid(
                     runtimeGraph,
-                    _targetPlatform,
-                    new[] { "linux", "linux-musl", "osx", "win", "freebsd", "illumos" },
+                    _targetRuntimeIdentifier,
```

Comment:
>             // Use the full target rid instead of just the target OS as the runtime graph
            // may only have the full target rid and not an OS-only rid for non-portable target rids
            // added by our source-build partners.

@jkoritzinsky @ViktorHofer The change does not match the comment. We were not passing the target OS before, we were passing the target rid with the architecture removed. So this could be `fedora.41` for example on a source-built SDK.

Reply:
The point of the change is to not pass an OS-only RID as our tooling to build a source-built SDK doesn't provide the OS-only RID (fedora.41 for example), only the OS+arch rid (fedora.41-x64).

I felt that the comment described this portion of the change well. If you think there's a better wording, we can change it.

Reply:
> The point of the change is to not pass an OS-only RID as our tooling to build a source-built SDK doesn't provide the OS-only RID (fedora.41 for example), only the OS+arch rid (fedora.41-x64).

Ah, that makes clear why the change is needed to make the lookup work.

Target OS made me think of https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/Unified-Build-Controls.md#output-controls.